synapse/lib/nexus.py

```diff
@@ -379,6 +379,13 @@ async def setindex(self, indx):
     async def _eat(self, item, indx=None):
 
         if self.donexslog:
+            nexsiden, event = item[:2]
```

Comment:
to reduce the amount of overhead while the nexus lock is held, could these be checked in `eat()` before we obtain the lock and fire the task?

Reply:
Good point, yeah that's a better place for it. Moved it up there.